alws/crud/repo_exporter.py

```diff
@@ -0,0 +1,143 @@
+    async with db.begin():
+        et_inserted = await db.execute(
+            insert(
+                models.ExportTask).values(
+                name=export_name,
+                status=0))
+        export_task_pk = et_inserted.inserted_primary_key[0]
+        response = await db.execute(query)
+        await db.commit()
+    for repo in response.scalars().all():
+        export_path = str(Path(settings.pulp_export_path,
+                               generate_repository_path(
+                                   export_task_pk, repo.name,
+                                   repo.arch, repo.debug)))
+        fs_exporter_href = await pulp_client.create_filesystem_exporter(
+            repo.name, export_path)
+        export_repos.append({
+            'path': export_path,
+            'exported_id': export_task_pk,
+            'repository_id': repo.id,
+            'fs_exporter_href': fs_exporter_href
+        })
+    if export_repos:
+        async with db.begin():
+            await db.execute(
+                insert(models.RepoExporter), export_repos)
+            await db.commit()
+    return export_task_pk
+
+
+async def execute_pulp_exporters_to_fs(db: Session,
+                                       export_id: int):
+    pulp_client = PulpClient(settings.pulp_host, settings.pulp_user,
+                             settings.pulp_password)
+    now = datetime.datetime.now()
+    query = select(models.RepoExporter.fs_exporter_href,
+                   models.RepoExporter.path,
+                   models.Repository.pulp_href).where(
+        models.RepoExporter.exported_id == export_id).join(
+        models.Repository).filter(
+        models.RepoExporter.repository_id == models.Repository.id)
+    async with db.begin():
+        await db.execute(
+            update(models.ExportTask).where(
+                models.ExportTask.id == export_id).values(exported_at=now,
+                                                          status=1))
+        response = await db.execute(query)
+        await db.commit()
+    exported_paths = []
+    for fs_exporter_href, fse_path, pulp_href in response:
+        latest_version_href = await pulp_client.get_repo_latest_version(
+            pulp_href)
+        fse_task = await pulp_client.export_to_filesystem(
+            fs_exporter_href, latest_version_href)
+        await pulp_client.wait_for_task(fse_task['task'])
+        exported_paths.append(fse_path)
+        await pulp_client.delete_filesystem_exporter(fs_exporter_href)
+    async with db.begin():
+        await db.execute(
+            update(models.ExportTask).where(
+                models.ExportTask.id == export_id).values(exported_at=now,
+                                                          status=3))
+        await db.commit()
+    return exported_paths
+
+
+async def create_filesystem_exporter(name : str, path: str) -> str:
+    pulp_client = PulpClient(settings.pulp_host, settings.pulp_user,
```

Comment:
There is unnecessary `pulp_client` instances in this functions, because you declare pulp instance in `execute_pulp_exporters_to_fs()`

Reply:
Don't understand the connection between these functions and pulp_client instance. Both functions are used **pulp_client.create_filesystem_exporter** that's true.

Reply:
Sorry, i'm doesn't notice that you use this functions in `routers.py`